client/src/components/workspaces/utils.ts

```diff
@@ -79,10 +86,42 @@ export async function workspaceShareClick(workspace: WorkspaceInfo, informationM
   await modal.onWillDismiss();
 }
 
+export async function workspaceFavoriteClick(
+  workspace: WorkspaceInfo,
+  favorites: Array<string>,
+  informationManager: InformationManager,
+  storageManager: StorageManager,
+): Promise<void> {
+  if (favorites.includes(workspace.id)) {
+    favorites.splice(favorites.indexOf(workspace.id), 1);
+    informationManager.present(
+      new Information({
+        message: { key: 'WorkspacesPage.Favorite.removed', data: { workspace: workspace.currentName } },
+        level: InformationLevel.Info,
+      }),
+      PresentationMode.Toast,
+    );
+  } else {
+    favorites.push(workspace.id);
+    informationManager.present(
+      new Information({
+        message: { key: 'WorkspacesPage.Favorite.added', data: { workspace: workspace.currentName } },
+        level: InformationLevel.Success,
+      }),
+      PresentationMode.Toast,
+    );
+  }
+  await storageManager.storeComponentData<WorkspacesPageSavedFavorites>(WORKSPACES_PAGE_DATA_KEY, {
+    favoriteList: favorites.toString(),
```

Comment:
Can't store it as a list of string? We really have to serialize it?

Reply:
Yes, otherwise we get this error
```
DOMException: Failed to execute 'put' on 'IDBObjectStore': [object Array] could not be cloned.
```